### PR TITLE
Feature/identity and eligibility factory

### PR DESCRIPTION
**Pull request by mikeSmith08**
- Created core component for checking the incoming request against a person found from the DWP seed database by NINO. This currently isn't used, it will be plugged into the controller in a future PR.
 - The core logic here is similar to what was created in the Smart Stub if this looks a little familiar.
 - Upgraded the version of Spring Boot and Jackson in the Smoke Tests to stop Snyk warnings.
 - Documentation of how the response is built up will be moved to README.md in another PR.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtils.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+     * @param ucSeededValue     The seeded value
+     * @param requestValue      The request value
+     * @param verificationCheck The check to run against those values.
+     * @return NOT_HELD if the seeded value is null, NOT_SUPPLIED if the request value is null, MATCHED if the predicate result is true otherwise NOT_MATCHED.
+     */
+    public static VerificationOutcome determineVerificationOutcome(String ucSeededValue, String requestValue, BiPredicate<String, String> verificationCheck) {
+        if (ucSeededValue == null) {
+            return VerificationOutcome.NOT_HELD;
+        }
+        if (requestValue == null) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtils.java:24]: This check should be first.

File: api/src/main/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityAndEligibilityResponseFactory.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+@Component
+public class IdentityAndEligibilityResponseFactory {
+
+    private static final String NO_HOUSEHOLD_IDENTIFIER_PROVIDED = "";
+
+    /**
+     * Builds a response based on the following criteria.
+     * <ul>
+     * <li> Identity status is MATCHED if NINO, Surname and DOB match the request</li>
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityAndEligibilityResponseFactory.java]: This needs to be more thorough when added to README.md.

File: api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    void shouldDetermineVerificationOutcome(String seededValue, String requestValue, boolean predicateReturn, VerificationOutcome expectedOutcome) {
+        //Given
+        BiPredicate<String, String> biPredicate = (s, s2) -> predicateReturn;
+        //When
+        VerificationOutcome outcome = IdentityVerificationUtils.determineVerificationOutcome(seededValue, requestValue, biPredicate);
+        //Then
+        assertThat(outcome).isEqualTo(expectedOutcome);
+    }
+
+    private static Stream<Arguments> determineVerificationOutcomeParams() {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java]: Some of these are now wrong when code is fixed.

File: api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    @Test
+    void shouldVerifyAreNotEqualIgnoringWhitespace() {
+        assertThat(IdentityVerificationUtils.areEqualIgnoringWhitespace("b a", "a b")).isFalse();
+    }
+
+    @Test
+    void shouldMatchFirstSixCharacters() {
+        assertThat(IdentityVerificationUtils.firstSixCharacterMatch("abcdefgh", "abcdefghyik")).isTrue();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java]: Add upper case char in here too.

File: api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    @Test
+    void shouldFailToMatchAdultWithDifferentDateOfBirth() {
+        PersonDTOV2 person = aPersonDTOV2WithDateOfBirth(MARGE_DOB);
+        assertThat(IdentityVerificationUtils.matchingAdult(HOMER, person)).isFalse();
+    }
+
+    @Test
+    void shouldMatchAdultWithNoDateOfBirthInUCAdult() {
+        UCAdult homer = aUCAdult(SIMPSON_SURNAME, HOMER_NINO_V2, null, HOMER_MOBILE, HOMER_EMAIL);
+        assertThat(IdentityVerificationUtils.matchingAdult(homer, aValidPersonDTOV2())).isTrue();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java]: Change the dob in aValidPersonDTOV2() from Homer's dob.

File: api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    @ParameterizedTest
+    @MethodSource("areEqualParams")
+    void shouldBeEqual(String string1, String string2) {
+        assertThat(IdentityVerificationUtils.areEqual(string1, string2)).isTrue();
+    }
+
+    @ParameterizedTest
+    @MethodSource("areNotEqualParams")
+    void shouldNotBeEqual(String string1, String string2) {
+        assertThat(IdentityVerificationUtils.areEqual(string1, string2)).isFalse();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/dwp/factory/v2/IdentityVerificationUtilsTest.java:98]: Add null tests.

**Review by YetAnotherMatt — APPROVED** [review]

**Review by gethyn1 — APPROVED** [review]

**Pull request merged**